### ISSUE TITLE
fixture.detectChanges() does not trigger ngOnChanges with updates made to fixture.point.componentInstance

### ISSUE THREAD

**Issue by Jonathan002** [ike18t/ng-mocks#2362]
### Description of the bug

`fixture.detectChanges()` does not work with the recommended practices of the docs.

The docs recommend to stay away from `fixture.componentInstance` and instead use `fixture.point.componentInstance`
or a reference to the `params` object used.

Link: https://ng-mocks.sudo.eu/api/MockRender#testing-changedetectionstrategyonpush#proxy-between-params-and-fixture

Image: 

<img width="1016" alt="Screen_Shot_2022-05-01_at_3_27_37_PM" src="https://user-images.githubusercontent.com/9410295/166167080-0e3383bf-fb22-45ed-acdc-48e24935a5c1.png">

### A repo with an example of the bug

I've created a demo repo here: https://github.com/Jonathan002/ngMocks-MockRender-detectChanges-example

Run `npm test` and `ngOnChanges` is only triggered when `fixture.componentInstance` props are updated. It makes it hard to follow the docs recommendations when detectChanges will only work with `fixture.componentInstance`

<img width="1328" alt="Screen_Shot_2022-05-01_at_3_56_33_PM" src="https://user-images.githubusercontent.com/9410295/166167942-4ee7db42-7545-4089-85a5-abdae6f6e5d9.png">

Thank you for looking into my issue

Link:

### Expected vs actual behavior

`fixture.detectChanges()` will trigger ngOnChanges in demo repo.

**Comment by satanTime** [comment]
Hi there,

Thanks for the report. I'll try to clarify that part better in the docs. 

An example to follow is here: https://ng-mocks.sudo.eu/guides/ngonchanges

**Comment by satanTime** [comment]
Hi @Jonathan002, I've updated docs. They should be deployed soon.

**Comment by satanTime** [comment]
[v13.5.1](https://github.com/ike18t/ng-mocks/releases/tag/v13.5.1) has been released and contains a fix for the issue. Feel free to reopen the issue or to submit a new one if you meet any problems.
